streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -230,38 +288,23 @@ public Sensor addLatencyAndThroughputSensor(final String scopeName,
                                                 final Sensor.RecordingLevel recordingLevel,
                                                 final String... tags) {
 
-        return addLatencyAndThroughputSensor(null,
-                                             scopeName,
-                                             entityName,
-                                             operationName,
-                                             recordingLevel,
-                                             tags);
-
-    }
-
-    public Sensor addLatencyAndThroughputSensor(final String taskName,
-                                                final String scopeName,
-                                                final String entityName,
-                                                final String operationName,
-                                                final Sensor.RecordingLevel recordingLevel,
-                                                final String... tags) {
-
         final Map<String, String> tagMap = constructTags(scopeName, entityName, tags);
         final Map<String, String> allTagMap = constructTags(scopeName, "all", tags);
 
         // first add the global operation metrics if not yet, with the global tags only
-        final Sensor parent = metrics.sensor(sensorName(buildUniqueSensorName(operationName, taskName), null), recordingLevel);
+        final Sensor parent = metrics.sensor(threadName + "." + operationName, recordingLevel);
         addLatencyMetrics(scopeName, parent, operationName, allTagMap);
         addThroughputMetrics(scopeName, parent, operationName, allTagMap);
 
         // add the operation metrics with additional tags
-        final Sensor sensor = metrics.sensor(sensorName(buildUniqueSensorName(operationName, taskName), entityName), recordingLevel, parent);
+        final Sensor sensor = metrics.sensor(entityName + "-" + (threadName + "." + operationName), recordingLevel, parent);
```

Comment:
I'm +1 to @mjsax 's comment: could we centralize all the logic of constructing the sensorName, metricsName, group, and the keyName of our own bookkeeping maps, so that they are not scattered and in danger of introducing bugs?